lib/run_loop/app.rb

```diff
@@ -69,20 +87,26 @@ def executable_name
 
     # Inspects the app's file for the server version
     def calabash_server_version
-      if valid?
-        path_to_bin = File.join(path, executable_name)
-        xcrun ||= RunLoop::Xcrun.new
-        hash = xcrun.exec(["strings", path_to_bin])
-        unless hash.nil?
-          version_str = hash[:out][/CALABASH VERSION: \d+\.\d+\.\d+/, 0]
-          unless version_str.nil? || version_str == ""
-            server_ver = version_str.split(":")[1].delete(' ')
-            RunLoop::Version.new(server_ver)
-          end
+      version = nil
+      executables.each do |executable|
+        version = strings(executable).server_version
+        break if version
+      end
+      version
+    end
+
+    # @!visibility private
+    # Collects the paths to executables in the bundle.
+    def executables
+      executables = []
+      Dir.glob("#{path}/**/*") do |file|
+        next if File.directory?(file)
+        next if skip_executable_check?(file)
```

Comment:
An optimization.

Calling `otool` on each file took ~2.0 seconds for a small bundle.  After applying this filter, it took ~0.09 seconds.